### ISSUE TITLE
Don't run the hook when nothing in the project root has changed

### ISSUE THREAD

**Issue by mjc** [qgadrian/elixir_git_hooks#107]
Once #106 is merged, when your elixir app is inside a subdirectory, it will always run even if what is changed is outside your elixir project.

On `pre-push`, provided hooks should make use of the commit information provided by git on `stdin` to check if any of the commits being pushed touch things inside the project root.

I don't think it's appropriate to check only `{lib,test}` as it's possible to put code outside that location.

We also should not start `mix` to determine this.

```
commits=$(git rev-list -n 1 "$range" -- $project_path)
if [ "$commits" != "" ]
then
  ...
fi
```

this pseudocode should mostly work.

`pre-push.sample` shows how to get `$range` populated correctly.

**Comment by qgadrian** [comment]
I will close this issue as this a nice to have and I don't have the capacity to maintain this feature.
